### ISSUE TITLE
TravisCI keeps running even after command fails

### ISSUE THREAD

**Issue by travisdowns** [travisdowns/uarch-bench#44]
Due to a [bug or mis-design](https://github.com/travis-ci/travis-ci/issues/1066) in TravisCI, the build will keep running even if a `script` command fails. For example, if the build fails, we'll still try to run `uarch-bench`. 

For now, we can probably include `set -e` to fix this in the absence of a configuration key in TravisCI to specify fail-fast behavior.

**Comment by nemequ** [comment]
It's a bit silly, but does it really matter if `./uarch-bench` is run? If compilation fails `./uarch-bench` won't exist, so the build will still fail…

Unless I'm missing something, it could also be fixed with

```diff
diff --git a/.travis.yml b/.travis.yml
index cdd4809..a97657e 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -27,5 +27,4 @@ matrix:
           - nasm
 
 script:
-- make CC=${C_COMPILER} CXX=${CXX_COMPILER} USE_LIBPFC=0 -j
-- ./uarch-bench --test-name=default/*
+- make CC=${C_COMPILER} CXX=${CXX_COMPILER} USE_LIBPFC=0 -j && ./uarch-bench --test-name=default/*
```

Which, while ugly, is less magical than `set -e`, and so would be my preference.

**Comment by travisdowns** [comment]
I don't matter _a lot_, no, but I want things to fail fast. The main annoyance is that if all the later steps run you get a bunch of confusing output. Then you have to keep scrolling up through all the failures to find the root failure. We should just stop at the root failure, which makes this all trivial. Granted, the amount of bogus lines currently isn't much, but it will get worse as I'm adding more stuff to the `script`.

Chaining the commands with `&&` works too, but doesn't exactly scale and has other limitations as well (e.g., you can't compose script recipes using references, etc). The error output from Travis is less useful too, since it will say "command A && B && C && D && ... failed" rather than just "command B failed".

Most importantly, it seems that `set -e` is a recommended way to do this is TravisCI, if you follow the above thread and related references (the `&&` is also a recommended way) - they even added a fix to make it work properly.

**Comment by travisdowns** [comment]
Fixed in bcf94fdc32 by adding a `set -e` at the start of the script entry in `.travis.yml`.
